src/SFML/Window/OSX/HIDInputManager.hpp

```diff
@@ -103,21 +93,64 @@ class HIDInputManager : NonCopyable
     ///
     /// Return sf::Keyboard::Unknown if it doesn't match any 'localized' keys.
     ///
-    /// By 'localized' I mean keys that depend on the keyboard layout
-    /// and might not be the same as the US keycode in some country
-    /// (e.g. the keys 'Y' and 'Z' are switched on QWERTZ keyboard and
+    /// By 'localized' we mean keys that depend on the keyboard layout
+    /// and might not be the same as the US keycode for some countries
+    /// (e.g. the keys 'Y' and 'Z' are swapped on QWERTZ keyboard and
     /// US keyboard layouts.)
     ///
     ////////////////////////////////////////////////////////////
-    static Keyboard::Key localizedKeys(UniChar ch);
+    static Keyboard::Key localizedKey(UniChar ch);
+
+    ////////////////////////////////////////////////////////////
+    /// Opposite transformation as localizedKeys
+    ///
+    /// Return 0x00 (NULL) for non-convertible keys/numpad numbers.
+    /// For letters, uppercase codes are returned.
+    /// Some returned value are specific to macOS.
+    ///
+    ////////////////////////////////////////////////////////////
+    static UniChar toUnicode(Keyboard::Key key);
+
+    ////////////////////////////////////////////////////////////
+    /// Try to convert a virtual keycode (HID level) into a
+    /// SFML scancode.
+    ///
+    /// Return sf::Keyboard::ScanUnknown if the keycode is unknown.
+    ///
+    ////////////////////////////////////////////////////////////
+    static Keyboard::Scancode nonLocalizedKey(UniChar virtualKeycode);
+
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \copydoc sf::Keyboard::isKeyPressed(Key)
+    ///
+    ////////////////////////////////////////////////////////////
+    bool isKeyPressed(Keyboard::Key key);
+
+    ////////////////////////////////////////////////////////////
+    /// \copydoc sf::Keyboard::isKeyPressed(Scancode)
+    ///
+    ////////////////////////////////////////////////////////////
+    bool isKeyPressed(Keyboard::Scancode code);
 
     ////////////////////////////////////////////////////////////
-    /// Try to convert a virtual keycode into a SFML key code.
+    /// \copydoc sf::Keyboard::localize
     ///
-    /// Return sf::Keyboard::Unknown if the keycode is unknown.
+    ////////////////////////////////////////////////////////////
+    Keyboard::Key localize(Keyboard::Scancode code);
+
+    ////////////////////////////////////////////////////////////
+    /// \copydoc sf::Keyboard::unlocalize
+    ///
+    ////////////////////////////////////////////////////////////
+    Keyboard::Scancode unlocalize(Keyboard::Key key);
+
+    ////////////////////////////////////////////////////////////
+    /// \copydoc sf::Keyboard::localizedRepresentation
     ///
     ////////////////////////////////////////////////////////////
-    static Keyboard::Key nonLocalizedKeys(UniChar virtualKeycode);
+    String localizedRepresentation(Keyboard::Scancode code);
```

Comment:
Two things:
* SFML convention uses `get` prefix for getters, not just the noun
* Is this the same as `sf::Keyboard::getDescription()`? In that case, we should probably give it the same name (but wait for renaming discussion of `getDescription()` to be resolved).

Reply:
It's an internal implementation and the naming is a leftover from the original naming (see https://github.com/SFML/SFML/pull/1235#discussion_r127936554), I'll adjust it.

Reply:
Changing it to `getDescription`